sdk/appcenter-storage/src/main/java/com/microsoft/appcenter/storage/models/BaseOptions.java

```diff
@@ -0,0 +1,43 @@
+package com.microsoft.appcenter.storage.models;
+
+import java.util.Calendar;
+
+public abstract class BaseOptions {
+
+    /**
+     * Cache does not expire
+     */
+    public static final int INFINITE = -1;
+    /**
+     * Do not cache documents locally
+     */
+    public static final int NO_CACHE = 0;
+    /**
+     * Default caching value of one hour
+     */
+    public static final int DEFAULT_ONE_HOUR = 60 * 60;
+    private int mTtl;
+
+    protected BaseOptions() {
+        this(DEFAULT_ONE_HOUR);
+    }
+
+    protected BaseOptions(int ttl) {
+        if (ttl < -1) throw new IllegalArgumentException("Time-to-live should be greater than or equal to zero, or -1 for infinite.");
```

Comment:
I think this should fall back to the default value and log an error instead of throwing here

Reply:
I'm not sure about it. If someone wants to instantiate a class with an incorrect parameter, they should know what they tried to instantiate is incorrect. Swapping the value at runtime and not letting them know is equivalent to just ignoring the parameter and doing something unrelated.

Reply:
Please commit via Android Studio with the following options, this is not formatted correctly here:

<img width="292" alt="screen shot 2019-03-07 at 10 57 40 am" src="https://user-images.githubusercontent.com/623672/53981666-e23f9c80-40c7-11e9-90b8-f7e3b44640ab.png">

Also for single line if we still use curly braces.

Reply:
As for the exception itself, it must be caught and logged before the public method called by user returns yes.